libraries/holochain-service-client/src/main/java/org/holochain/androidserviceruntime/holochain_service_client/HolochainServiceClient.kt

```diff
@@ -49,42 +50,126 @@ class HolochainServiceClient(
     }
 
     /// Install an app
-    fun installApp(payload: InstallAppPayloadFfiParcel): AppInfoFfiParcel {
-        // Call installApp on service
-        return this.mService!!.installApp(payload)
+    suspend fun installApp(payload: InstallAppPayloadFfiParcel): AppInfoFfiParcel {
+        Log.d(TAG, "installApp")
+
+        val deferred = CompletableDeferred<AppInfoFfiParcel>()
```

Comment:
A `CompletableDeferred` is akin to `Promise` in javascript. This is converting a callback to a promise.